app/services/recurring_contribution/subscriptions/update_status.rb

```diff
@@ -0,0 +1,21 @@
+class RecurringContribution::Subscriptions::UpdateStatus
+  def initialize(subscription_id, current_status)
+    @juntos_subscription = Subscription.find_by(subscription_code: subscription_id)
+    @current_status = current_status
+  end
+
+  def self.process(subscription_id, params)
+    new(subscription_id, params).process
+  end
+
+  def process
+    update_juntos_subscription_status
```

Comment:
I don't see any problem if you use the logic of `update_juntos_subscription_status` direct inside of `process`, instead of calling `update_juntos_subscription_status`.
What do you think?